### ISSUE TITLE
Export SAM template outputs

### ISSUE THREAD

**Issue by crawforde** [aws/chalice#1087]
I noticed the generated SAM templates have some very useful output values, such as the EndpointURL. It would be really great if these were exported  with a unique name, such as `"${AWS::StackName}-LogicalID"` so we could import them into other cloudformation templates (such as a frontend stack making requests to the Chalice api). Right now, it seems like the only way to get at them programmatically is to write a script using the aws command line tools to describe the stack and parse the result for outputs. A benefit of importing values directly into other CF templates is also that AWS will then restrict any actions that would change these exported values and break these other stacks.

**Comment by kyleknap** [comment]
I agree. Having the ability to export these output values would be very useful. Another option (which I tend to do) in this situation is write a wrapper script that after running `chalice package`  will take the generated CloudFormation and inject any extra that I want (i.e. CloudFormation exports). We currently have a feature request tracking better deployment hooks: https://github.com/aws/chalice/issues/758 so you would not have to do a one-off wrapper script after running the `chalice package` command. However, having built-in support for adding exports to the generated CloudFormation template is useful by itself. Marking as a feature request.

**Comment by brianwhigham** [comment]
+1

**Comment by nikolasten** [comment]
+1

**Comment by silverjam** [comment]
:+1:

**Comment by smitrob** [comment]
+1
